main.cpp

```diff
@@ -70,29 +70,412 @@ write 3 UDTs below that EACH have:
+        if(containersLoaded == numOfCargoContainers)
+        {
+            std::cout << "ship is ready for departure" << std::endl;
+            return true;
+        }
+    }
+    return false;
+}
 /*
  copied UDT 3:
  */
+struct Computer
+{
+    double storageInGb {500.0};
+    float processorSpeed {3.5f};
+    float displaySize {13.0f};
+    int displayBrightness {80};
+    std::string manufacturer {"Apple"};
+    Computer();
+    ~Computer();
+
+    void storeData(int numFiles, int fileSize, std::string fileName = "unknown");
+    void processData(float amountToProcess, float timeToProcess = 0.02f);
+    void displayData(std::string fileName = "unknown", bool fullscreen = false);
+    void changeDisplayBrightness(int newBrightVal);
+};
+
+Computer::Computer()
+{
+    std::cout << "Computer being constructed!" << std::endl;
+}
 
+Computer::~Computer()
+{
+    std::cout << "Computer destroyed" << std::endl;
+}
+
+void Computer::storeData(int numFiles, int fileSize, std::string fileName)
+{
+    std::string database = "added " + fileName;
+    storageInGb -= numFiles * fileSize;
+    std::cout << "Computer::storeData() " << database << std::endl;
+}
+
+void Computer::processData(float amountToProcess, float timeToProcess)
+{
+    processorSpeed = amountToProcess * timeToProcess;
+    std::cout << "Computer::processData() speed: " << processorSpeed << std::endl;
+}
+
+void Computer::displayData(std::string fileName, bool fullscreen)
+{
+    if (!fullscreen)
+        displaySize *= 0.5f;
+
+    std::string displayText = "file contains: " + fileName;    
+    std::cout << "Computer::displayData(): " << displayText << std::endl;
+}
+
+void Computer::changeDisplayBrightness(int newBrightVal)
+{
+    if(newBrightVal > displayBrightness)
+    {
+        for(; displayBrightness < newBrightVal; ++displayBrightness)
+        {
+            std::cout << "increasing brightness: " << displayBrightness << std::endl;
+        }
+    }
+    else
+    {
+        for(; displayBrightness > newBrightVal; --displayBrightness)
+        {
+            std::cout << "decreasing brightness: " << displayBrightness << std::endl;
+        }
+    }  
+}
 /*
  new UDT 4:
  with 2 member functions
  */
+struct CoffeeShop
+{
+    CoffeeShop();
+    ~CoffeeShop();
+
+    CoffeeMachine coffeeMachine1;
+
+    void prepareOrder(int cupsOrdered, int brewStrength);
+    void cleanMachine(int cyclesMade);
 
+};
+
+CoffeeShop::CoffeeShop()
+{
+    std::cout << "CoffeeShop being constructed" << std::endl;
+}
+
+CoffeeShop::~CoffeeShop()
+{
+    std::cout << "CoffeeShop destroyed" << std::endl;
+}
+
+void CoffeeShop::cleanMachine(int cyclesMade)
+{
+    if(coffeeMachine1.brewSettings.flashDescalingIndicator(3, cyclesMade))
+    {
+        std::cout << "starting descaling program for " << coffeeMachine1.brandName << " machine 1 ..." << std::endl;
+        std::cout << "resetting cycle counter... program finished!" << std::endl;
+    }
+    else
+        return;
```

Comment:
This `else` does nothing
this function returns `void`
remove it.
- [ ] revise